### PR TITLE
Extra meta

### PR DESCRIPTION
**Pull request by wheresrhys**
A couple of approaches for tackling
a) stateful cta's
b) extra metadata (as needed by typeahead)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .PHONY: test
 
-test: 
+test:
 	#origami-build-tools verify
```
Comment (commuterjoy) [inline Makefile]: oops. i'll fix this. shouldn't be commented out.

File: src/analytics/cta.js [hunk 2]
```diff
@@ -16,16 +16,22 @@ CTA.prototype.track = function (root) {
...
 
 		var domPath = getDomPath(e.target, []);
+		var targetPrefix = e.target.hasAttribute('aria-pressed') ? 'un-' : '';
+
+		domPath[0] = targetPrefix + domPath[0];
+
+		meta.nodeName = e.target.nodeName.toLowerCase();
+		meta.domPath = domPath.reverse().join(" | ");
+		meta.domPathTokens = domPath;
+		meta.target = domPath[domPath.length -1];
```
Comment (commuterjoy) [inline src/analytics/cta.js]: did you change this because of a bug?

File: src/analytics/cta.js [hunk 3]
```diff
@@ -16,16 +16,22 @@ CTA.prototype.track = function (root) {
...
 		}
 
-		// FIXME introduce a 200ms delay in to hyperlinks - if !sendBeacon
+		var meta = e.target.getAttribute('data-trackable-meta');
+		meta = meta ? JSON.parse(meta) : {};
 
 		var domPath = getDomPath(e.target, []);
+		var targetPrefix = e.target.hasAttribute('aria-pressed') ? 'un-' : '';
+
+		domPath[0] = targetPrefix + domPath[0];
```
Comment (commuterjoy) [inline src/analytics/cta.js]: Can we have a separate property called `domPathAria` whilst we test this out?

File: README.md [hunk 4]
```diff
@@ -156,13 +156,17 @@ A click on the SPAN element will return `menu | goodbye`.
 The attribute `data-trackable-terminate` can be used to stop the tracking code
 collecting trackable attributes beyond a given node.
 
+The attribute `data-trackable-meta`, containing a JSON string, can be used to send additional information in the event detail -properties of the JSON will be added as properties of the event detail
+
+If the attribute `aria-pressed` exists on the target 'un-' will be prefixed to the data-trckable value, so e.g. the button above woudl have the dom path `menu | item 1 | un-hello`
```
Comment (commuterjoy) [inline README.md]: woudl -> would

**Comment by commuterjoy** [comment]
@wheresrhys  Do browsers set the state of aria-pressed automatically ?

File: src/analytics/cta.js (hunk 3 above)
Comment (wheresrhys) [inline src/analytics/cta.js]: yep

File: src/analytics/cta.js (hunk 2 above)
Comment (wheresrhys) [inline src/analytics/cta.js]: nope, in order to only have to set 'un-' in one place

**Comment by wheresrhys** [comment]
Nope, but both next-myft-ui and o-expander (the only stateful buttons we have I think) do

**Pull request merged**